addons/pos_loyalty/static/src/overrides/models/pos_store.js

```diff
@@ -359,6 +359,26 @@ patch(PosStore.prototype, {
         }
         return couponList;
     },
+    getCachedCoupon(programId, partnerId) {
+        for (const coupon of Object.values(this.couponCache)) {
```

Comment:
I'm also wondering if this method could be called without a partner selected. This could lead to returning a non-nominative coupon. Would that be bad ?